src/components/tree-item/tree-item.tsx

```diff
@@ -425,6 +430,14 @@ export class TreeItem
   //
   //--------------------------------------------------------------------------
 
+  /**
+   * Used to make sure initially expanded tree-item can properly
+   * transition and emit events from closed state when rendered.
+   *
+   * @private
+   */
+  @State() transitionListenersAddedOnInitialLoad = false;
```

Comment:
WDYT about:

1. renaming this to `updateAfterInitialRender`
1. removing the line in `componentWillLoad` that sets it
1. adding a `ref` to host or the expanding container that sets the flag if `expanded === true`?

I think we should at least do 1 even if it's with another name. Reason being that there's no direct connection between the transition listeners being added and how/when the flag is set.